fvm/transactionInvoker.go

```diff
@@ -358,9 +359,22 @@ func (executor *transactionExecutor) normalExecution() (
 	err error,
 ) {
 	var maxTxFees uint64
+	var frozenAccounts map[flow.Address]struct{}
 	// run with limits disabled since this is a static cost check
 	// and should be accounted for in the inclusion cost.
 	executor.txnState.RunWithMeteringDisabled(func() {
+		frozenAccounts, err = executor.env.GetFrozenAccounts()
+		if err != nil {
+			err = fmt.Errorf("getting frozen accounts failed: %w", err)
+			return
+		}
+
+		// if payer is frozen, fail the transaction early
+		if _, ok := frozenAccounts[executor.proc.Transaction.Payer]; ok {
+			err = errors.NewAccountFrozenError(executor.proc.Transaction.Payer)
+			return
+		}
```

Comment:
Should this only be the payer? I assumed we would freeze the authorizers

Reply:
any edit to the account is frozen by the other part of the code. this is a special case I added for spam protection.

If the payer is frozen they wont be able to pay for running the transaction, so dont run it at all.